test/performance/compiler/elliot/passCheck.prediff

```diff
@@ -1,32 +1,64 @@
+#
+# To fix compiler performance testing, the .dat files and .graph files have to
+# be updated even for simple things like moving when a pass is run. The keys
+# and data in the .dat have to be in the same order as the passes run. New
+# passes have to have old dates backfilled with a '-'. The .graph file keys
+# should also be kept in the same order.
+#
+# Unfortunately it's hard to automate adding/reordering/removing passes. At
+# least for anything even a little complex. e.g. How do you automatically tell
+# the difference between if a pass was removed and a new one was added vs
+# simple renaming? For now, just ping Elliot, and he'll update the right files.
+# External contributers: just make a note of this failure in your pull request.
+#
+# This isn't meant to hold up development and if nobody who knows how to update
+# the compiler performance .dat and .graph files is around, just update
+# compilerPerformance.perfkeys so there's no nightly regressions and make sure
+# somebody eventually changes the .dat and .graph files. It's not the worst
+# thing to lose a few nights of compiler performance data.
 
-logfile = sys.argv[2]
-tmplogfile = logfile + '.tmp'
+import sys
 
-file = open(logfile, 'r')
-tmpfile = open(tmplogfile, 'w')
+logfile = sys.argv[2]
 
-# remove all of the actual times from --print-passes
-# It outputs passname : <time for pass> and we want
-# to just remove <time for pass>
-lines = file.readlines()
-for line in lines:
-  splitline = line.split(':')[0].strip() + ' :'
-  tmpfile.write(splitline + '\n')
-  # --print-passes prints detailed output after printing the total time for
-  # each pass. This just strips all that output which occurs after the total
-  # is printed.
-  if line.strip().startswith('total time :') :
-    break
+loglines = []
+with open(logfile, 'r') as f:
+    loglines = f.readlines()
 
-file.close()
-tmpfile.close()
+# remove all of the actual times from --print-passes. It outputs
+# '<passname> : <time for pass>' and we want to remove <time for pass>
+with open(logfile, 'w') as f:
+    for line in loglines:
+      splitline = line.split(':')[0].strip() + ' :'
+      f.write(splitline + '\n')
+      # --print-passes prints detailed output after printing the total
+      # time for each pass. This just strips all that output which
+      # occurs after the total is printed.
+      if line.strip().startswith('total time :') :
+        break
 
-os.rename(tmplogfile, logfile)
 
 # copy the compilerPerformance perf keys as the .good file for this test. That
-# way we don't have to update the .good and the perfkeys
-shutil.copy('../compilerPerformance.perfkeys', 'passCheck.good')
+# way we don't have to update the .good and the perfkeys. Ignore comments and
+# empty lines in the compilerPerformance.perfkeys file
+passes = []
+with open('../compilerPerformance.perfkeys', 'r') as f:
+    passes = f.readlines()
+passes = [ p for p in passes if p.strip() and not p.strip().startswith('#') ]
```

Comment:
I don't know if I'd call it expensive. It could be faster, but it's a really small input size you'll never notice

Still using a list comprehension but eliminating the second strip() would be: 

```python
# note this is probably more expensive than the current version
passes = [ p.strip() for p passes ]
passes = [ p for p in passes if p and not p.startswith('#') ]
```


Could also do something like: 

```python
passes = []
with open('../compilerPerformance.perfkeys', 'r') as f:
    for line in f:
        line = line.strip()
        if not line.startswith('#'):
            passes.append(line)
```

I'm not opposed to any of these versions. I don't think speed is a concern at all, but if one is easier to read, it could be worthwhile.